Spigot-Server-Patches/0649-Entity-load-save-limit-per-chunk.patch

```diff
@@ -32,7 +32,7 @@ index c5495e02c3fe271b26f62ea2ec64e07957edf37e..234d2daecc5d0bf6a99c0a5f4a87f947
 +        getInt("entity-per-chunk-save-limit.ender_pearl", -1);
 +        getInt("entity-per-chunk-save-limit.arrow", -1);
 +        EntityTypes.getEntityNameList().forEach(name -> {
-+            final EntityTypes<?> type = EntityTypes.getByName(name.getKey()).orElseThrow(() -> new IllegalStateException("Unknown Entity Type: " + name.toString()));
++            final EntityTypes<?> type = EntityTypes.getByName(name.toString()).orElseThrow(() -> new IllegalStateException("Unknown Entity Type: " + name.toString()));
 +            final String path = ".entity-per-chunk-save-limit." + name.getKey();
```

Comment:
I am somewhat concerned over the effect this has. Two entities, `custom:creeper` and `minecraft:creeper` will have the same limits in this case. Is this something we should perhaps look into?

Reply:
Shouldn't we be more open?

Reply:
Ideally, this would only register for `minecraft:creeper` unless a namespace has been specified. That'd let you do something like the following:

```yaml
"custom:creeper": 13
creeper: 7 # or "minecraft:creeper"
```

Reply:
If' i'm understanding this code correctly, `name` _is_ the key (`minecraft:creeper`), so `name.getKey()` is returning `creeper`. This change is so `name.toString()` returns `minecraft:creeper` (or `custom:creeper` since he's adding his own mob).

Reply:
That is true, yes, but I commented on the line below the change: `final String path = ".entity-per-chunk-save-limit." + name.getKey();`. This means you have the same key issue but with another property.

Reply:
Oof. didn't even see that ^_^